packages/headless/src/features/insight-user-actions/insight-user-actions-preprocessing.test.ts

```diff
@@ -0,0 +1,1152 @@
+          searchHub: 'in-product-help',
+          document: {title: 'title', uriHash: 'TtnKwc0Lo2GY9WAi'},
+        },
+        {
+          actionType: 'CUSTOM',
+          timestamp: new Date('Fri Apr 01 2022 14:17:00 GMT').valueOf(),
+          eventData: {type: 'useless_event', value: ''},
+          searchHub: 'community-support',
+          document: {},
+        },
+        {
+          actionType: 'CLICK',
+          timestamp: new Date('Fri Apr 01 2022 14:15:00 GMT').valueOf(),
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {title: 'title', uriHash: 'KXñi9EWk38wnb1tt'},
+        },
+        {
+          actionType: 'SEARCH',
+          timestamp: new Date('Fri Apr 01 2022 14:13:00 GMT').valueOf(),
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {},
+          query: '',
+        },
+        {
+          actionType: 'VIEW',
+          timestamp: new Date('Thu Mar 31 2022 16:35:00 GMT').valueOf(),
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {
+            contentIdKey: 'sftitle',
+            contentIdValue: 'Blaze pair with iPhone not working',
+          },
+        },
+        {
+          actionType: 'CLICK',
+          timestamp: new Date('Thu Mar 31 2022 16:32:00 GMT').valueOf(),
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {title: 'title', uriHash: 'caCgiG2JPzjZfS7G'},
+        },
+        {
+          actionType: 'CUSTOM',
+          timestamp: new Date('Thu Mar 31 2022 16:31:30 GMT').valueOf(),
+          eventData: {
+            type: 'smartSnippetSuggestions',
+            value: 'expandSmartSnippetSuggestion',
+          },
+          searchHub: 'in-product-help',
+          document: {},
+        },
+      ];
+      const mockRawActions = [...fakeActions].slice(0, 8);
+      const mappedAndSortedActions =
+        mapAndSortActionsByMostRecent(mockRawActions);
+
+      expect(mappedAndSortedActions).toEqual(expectedMappedAndSortedActions);
+      expect(mappedAndSortedActions[0].timestamp).toBeGreaterThan(
+        mappedAndSortedActions[1].timestamp
+      );
+    });
+
+    describe('when a rawUserAction cannot be parsed', () => {
+      let consoleWarnSpy: jest.SpyInstance;
+
+      it('should log a warning in the console and ignore that value', () => {
+        const invalidRawUserAction = {
+          name: 'Custom',
+          value: '{ "key": "value",,,, }',
+          time: '2023-10-01T12:00:00Z',
+        };
+
+        consoleWarnSpy = jest.spyOn(console, 'warn').mockImplementation();
+
+        const mappedAndSortedActions = mapAndSortActionsByMostRecent([
+          invalidRawUserAction,
+        ]);
+
+        expect(mappedAndSortedActions.length).toEqual(0);
+        expect(mappedAndSortedActions).toEqual([]);
+
+        expect(consoleWarnSpy).toHaveBeenCalledTimes(1);
+
+        consoleWarnSpy.mockRestore();
+      });
+    });
```

Comment:
To keep the test clean:

```suggestion
    describe('when a rawUserAction cannot be parsed', () => {
      let consoleWarnSpy: jest.SpyInstance;

      beforeAll(() => {
        consoleWarnSpy = jest.spyOn(console, 'warn').mockImplementation();
      });

      afterAll(() => {
        consoleWarnSpy.mockRestore();
      });

      it('should log a warning in the console and ignore that value', () => {
        const invalidRawUserAction = {
          name: 'Custom',
          value: '{ "key": "value",,,, }',
          time: '2023-10-01T12:00:00Z',
        };

        const mappedAndSortedActions = mapAndSortActionsByMostRecent([
          invalidRawUserAction,
        ]);

        expect(mappedAndSortedActions.length).toEqual(0);
        expect(mappedAndSortedActions).toEqual([]);
        expect(consoleWarnSpy).toHaveBeenCalledTimes(1);
      });
    });
```